### PR TITLE
Event-driven example of cloud storage

### PR DESCRIPTION
**Pull request by byam**
Event-driven example of cloud storage

### CODE REVIEW

File: gcs-triggered/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+## Event-Driven 
+
+1. Upload a file to Cloud Storage bucket
+2. Object Notification send to Cloud Pub Sub topic
```
Comment (Kenji-H) [inline gcs-triggered/README.md]: send -> is sent to

File: gcs-triggered/bin/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+# run pipeline
+mvn compile exec:java \
+  -Dexec.mainClass=com.byam.beam.examples.GcsTriggered\
+  -Dexec.args="\
+  --runner=${RUNNER} \
+  --project=${GOOGLE_PROJECT} \
+  --topic=projects/${GOOGLE_PROJECT}/topics/${TOPIC_NAME} \
+  " \
+  -Dexec.cleanupDaemonThreads=false
```
Comment (Kenji-H) [inline gcs-triggered/bin/deploy.sh]: Why did you set this to `false` ?

File: gcs-triggered/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+3. Cloud Dataflow
+    - Read Cloud Pub Sub topic with PubSubIO
+    - Parse object notification json to get file path
+    - Read Cloud Storage file with TextIO
+    - Count words
```
Comment (Kenji-H) [inline gcs-triggered/README.md]: Seems It's different from the actual behavior.

File: gcs-triggered/src/main/java/com/byam/beam/examples/Options.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.Validation.Required;
+import org.apache.beam.sdk.options.ValueProvider;
+
+public interface Options extends PipelineOptions {
+
+    @Description("Cloud Pub Sub Topic Name.")
+    @Required
+    ValueProvider<String> getTopic();
+    void setTopic(ValueProvider<String> value);
```
Comment (Kenji-H) [inline gcs-triggered/src/main/java/com/byam/beam/examples/Options.java]: I think it's ok to use `String` as an argument for setTopic.
What is the benefit of using `ValueProvider` here?

File: gcs-triggered/src/test/java/com/byam/beam/examples/GcsTriggeredTest.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    /**
+     * @return the suite of tests being tested
+     */
+    public static Test suite()
+    {
+        return new TestSuite( GcsTriggeredTest.class );
+    }
+
+    /**
+     * Rigourous Test :-)
```
Comment (Kenji-H) [inline gcs-triggered/src/test/java/com/byam/beam/examples/GcsTriggeredTest.java:32]: nice :)

**Review by Kenji-H — CHANGES_REQUESTED** [review]
@byam Added some comments. Please check them out.

File: gcs-triggered/src/main/java/com/byam/beam/examples/Options.java (hunk 4 above)
Comment (byam) [reply gcs-triggered/src/main/java/com/byam/beam/examples/Options.java]: It is useful for using Google Cloud Template. But i will change it to usual `string`. 
https://cloud.google.com/dataflow/docs/templates/creating-templates#using-staticvalueprovider

File: gcs-triggered/bin/deploy.sh (hunk 2 above)
Comment (byam) [reply gcs-triggered/bin/deploy.sh]: Avoiding local error. I will remove it next commit!

**Comment by byam** [comment]
@Kenji-H Fixed some codes!

**Review by Kenji-H — APPROVED** [review]
@byam Thanks. LGMT!

**Pull request merged**